vcs/gitcmd/repo.go

```diff
@@ -962,6 +962,35 @@ func (r *Repository) Committers(opt vcs.CommittersOptions) ([]*vcs.Committer, er
 	return committers, nil
 }
 
+func (r *Repository) ListFiles(at vcs.CommitID) ([]string, error) {
+	if err := checkSpecArgSafety(string(at)); err != nil {
+		return nil, err
+	}
+
+	r.editLock.RLock()
+	defer r.editLock.RUnlock()
+
+	if at == "" {
+		at = "HEAD"
+	}
+
+	cmd := exec.Command("git", "ls-files", "--with-tree", string(at))
+	cmd.Dir = r.Dir
+	out, err := cmd.Output()
+	if err != nil {
+		return nil, fmt.Errorf("exec `git ls-files --with-tree %v` failed: %v", at, err)
+	}
+	out = bytes.TrimSpace(out)
```

Comment:
> because `strings.Split("", "\n")` returns an empty slice

It does not return an empty slice. It returns a slice with 1 element, a blank string.

https://play.golang.org/p/M_wIOtBBg0